metadata_backend/api/operators.py

```diff
@@ -755,27 +752,29 @@ async def remove_object(self, folder_id: str, collection: str, accession_id: str
 
         LOG.info(f"Removing object {accession_id} from {folder_id} succeeded.")
 
-    async def delete_folder(self, folder_id: str) -> str:
+    async def delete_folder(self, folder_id: str) -> Union[str, None]:
         """Delete object folder from database.
 
         :param folder_id: ID of the folder to delete.
         :raises: HTTPBadRequest if deleting was not successful
         :returns: ID of the folder deleted from database
         """
-        try:
-            await self.check_folder_exists(folder_id)
-            delete_success = await self.db_service.delete("folder", folder_id)
-        except (ConnectionFailure, OperationFailure) as error:
-            reason = f"Error happened while deleting folder: {error}"
-            LOG.error(reason)
-            raise web.HTTPBadRequest(reason=reason)
-        if not delete_success:
-            reason = f"Deleting for {folder_id} from database failed."
-            LOG.error(reason)
-            raise web.HTTPBadRequest(reason=reason)
-        else:
-            LOG.info(f"Deleting folder with id {folder_id} to database succeeded.")
-            return folder_id
+        published = await self.db_service.published_folder(folder_id)
+        if not published:
+            try:
+                delete_success = await self.db_service.delete("folder", folder_id)
+            except (ConnectionFailure, OperationFailure) as error:
+                reason = f"Error happened while deleting folder: {error}"
+                LOG.error(reason)
+                raise web.HTTPBadRequest(reason=reason)
+            if not delete_success:
+                reason = f"Deleting for {folder_id} from database failed."
+                LOG.error(reason)
+                raise web.HTTPBadRequest(reason=reason)
+            else:
+                LOG.info(f"Deleting folder with id {folder_id} to database succeeded.")
+                return folder_id
+        return None
```

Comment:
Maybe this makes sense, but should/could the if-else be inside `try` ?

Reply:
good point, fixed